pkg/executor/default.go

```diff
@@ -74,24 +75,23 @@ func (l opList) uniqueNames() {
 	}
 }
 
-func (e *DefaultExecutor) applyStage(stage schema.Stage, fs vfs.FS, console plugins.Console) error {
+func (e *DefaultExecutor) applyStage(stageName string, stage schema.Stage, fs vfs.FS, console plugins.Console) error {
 	var errs error
 	for _, p := range e.conditionals {
 		if err := p(e.logger, stage, fs, console); err != nil {
 			e.logger.Warnf("(conditional) Skip '%s' stage name: %s",
-				err.Error(), stage.Name)
+				err.Error(), stageName)
```

Comment:
This diff with the actual Apply method was causing the DAG to be created properly with names generated if the stage had no name, but then on the log it will still shop as empty.